agent/Equalizer.py

```diff
@@ -0,0 +1,269 @@
+                choice = self.get_low_card(suits[3])
+            else:
+                if len(suits[2]) == 0:
+                    print("hi-follow")
+                choice = self.get_low_card(suits[2])
+        else:
+            # clubs or diamonds have been played, follow with our high card (if we can)
+            # we care less here about which to pick if we're void, so just go with the suit
+            # thats the smallest
+            if len(suits[1]) == 0 or len(suits[1]) > len(suits[0]) != 0:
+                choice = self.get_highest_safe_card(suits[0], partial_state)
+            else:
+                choice = self.get_highest_safe_card(suits[1], partial_state)
+        return choice
+
+    def sort_suits(self,
+                   partial_state: HeartsState):
+        """ Sort out the Cards by Suit and return a List of the suits"""
+        # Clubs - diamonds - Spades - Hearts
+        cards = []
+        clubs = []
+        diamonds = []
+        spades = []
+        hearts = []
+        card_index = 0
+        for i in partial_state.values:
+            if 0 < i < 5:
+                cards.append(card_index)
+            card_index = card_index + 1
+
+        for i in range(4):
+            start = i * 13
+            end = start + 13
+            for card in cards:
+                if i == 0:
+                    if start <= card < end:
+                        clubs.append(card)
+                elif i == 1:
+                    if start <= card < end:
+                        diamonds.append(card)
+                elif i == 2:
+                    if start <= card < end:
+                        spades.append(card)
+                elif i == 3:
+                    if start <= card < end:
+                        hearts.append(card)
+
+        suits = [clubs, diamonds, spades, hearts]
+        return suits
+
+    def played_cards_in_trick(self,
+                              partial_state: HeartsState):
+        """ Get the currently played cards in the trick"""
+        # why
+        trick_cards = []
+        card_index = 0
+        for i in partial_state.values:
+            if 20 < i < 35:
+                trick_cards.append(card_index)
+            card_index = card_index + 1
+        return trick_cards
+
+    def get_low_card(self,
+                     suit_to_choose_from: list):
+        """In this function we want to want to pick the lowest card from this suit"""
+        suit_to_choose_from.sort()
+        choice = suit_to_choose_from[0]
+        return choice
+    def passing(self, partial_state: HeartsState):
+        hand = self.sort_suits(partial_state)
+        # We want to void out Clubs - Diamonds - Spades - Hearts
+        pirority = [ 0 , 1, 2 ,3 ]
+        # Reference to all the amount of cards in each suit
+        suit_map = {
+                "D":len(hand[1]),
+                "C":len(hand[0]),
+                "S":len(hand[2]),
+                "H":len(hand[3])
+        }
```

Comment:
Minor things:
Probably should add a space between the end of the previous method and passing method.
Priority is misspelled.  ` "D":len(hand[1]), "C":len(hand[0]),` could be switched. 
Priority and suit_map aren't used all at in this method, is this a place holder for something else down the line?

Reply:
I'm not entirely sure.  This is Michael's code.

@msomarri

Reply:
Added in the spacing, fixed spelling, and commented out the unused variables.